lib/GenomeFileUtil/core/GenbankToGenome.py

```diff
@@ -516,15 +515,7 @@ def _save_assemblies(self, workspace_id, genome_objs):
             contigs = Bio.SeqIO.parse(genome_obj.consolidated_file, "genbank")
             genome_obj.assembly_id = f"{genome_obj.genome_name}_assembly"
             genome_obj.assembly_path = f"{self.cfg.sharedFolder}/{genome_obj.assembly_id}.fasta"
-
-            # populate contig_seq
-            genome_obj.extra_info = self._get_contigs_and_extra_info(
-                contigs, genome_obj
-            )
-
-            # Output as fasta file
-            contigs_output = Bio.SeqIO.parse(genome_obj.consolidated_file, "genbank")
```

Comment:
So we were parsing the file twice in this method previously?

Reply:
Yes, because Bio.SeqIO.parse() returns an iterator of contigs, which is passed into `self._get_contigs_and_extra_info()` (https://github.com/kbaseapps/GenomeFileUtil/blob/578814f0a10368dc9d30ad9b805ee0b94d8efd83/lib/GenomeFileUtil/core/GenbankToGenome.py#L521) for processing. Now, by using load_contig() at the beginning, we are parsing the file only once for bulk upload.

Reply:
@[MrCreosote](https://github.com/MrCreosote) MrCreosote Pending

I see, so we were exhausting the iterator and then creating a new iterator. Since we always handle the extra info early in the loop, we don't need to do it here?

Reply:
Right, that's why I isolate load_contigs and moved it up.

Reply:
:+1: